### ISSUE TITLE
Signal networks, clocked signal functions and resampling buffers are profunctors

### ISSUE THREAD

**Issue by turion** [turion/rhine#138]
I think that for any functor `m` and clocks `cl, cl'`, all `ClSF m cl`, `SN m cl` and `Resbuf m cl cl'` are instances of `Profunctor`. This means that we can pre- and postcompose them with pure functions, and that precomposition is contravariant. In arrows, precomposition is known as the [(^>>)](https://hoogle.haskell.org/?hoogle=%5E%3E%3E) operator.

## To do

* Add dependency of the `profunctor` package
* Implement the three mentioned `Profunctor` instances

**Comment by freckletonj** [comment]
I'm still wrapping my head around dunai/rhine, but if ClSF and SN are profunctors, shouldn't we more simply just make MSF a profunctor?

**Comment by turion** [comment]
That's absolutely right. If `MSF` were a profunctor, `ClSF` would automatically become one. `SN` would require a separate instance, but it's not hard.

See also:

https://github.com/ivanperez-keera/dunai/issues/224

**Comment by turion** [comment]
Similarly, I believe that at least `ClSF`s are [`Traversing`](https://hackage.haskell.org/package/profunctors-5.6.2/docs/Data-Profunctor-Traversing.html#t:Traversing). See https://github.com/ivanperez-keera/dunai/issues/224#issuecomment-825053518.
